src/uct/api/uct.h

```diff
@@ -226,9 +226,12 @@ typedef struct uct_tl_resource_desc {
                                                        is called. */
 
         /* Event notification */
-#define UCT_IFACE_FLAG_EVENT_SEND_COMP    UCS_BIT(46) /**< Event notification of send completion is supported */
-#define UCT_IFACE_FLAG_EVENT_RECV_AM      UCS_BIT(47) /**< Event notification of active message receive is supported */
-#define UCT_IFACE_FLAG_EVENT_RECV_SIG_AM  UCS_BIT(48) /**< Event notification of signaled active message is supported */
+#define UCT_IFACE_FLAG_EVENT_SEND_COMP UCS_BIT(46) /**< Event notification of send completion is
+                                                        supported */
+#define UCT_IFACE_FLAG_EVENT_RECV      UCS_BIT(47) /**< Event notification of tag and active message
```

Comment:
tag and active message ? Or tag or active message ? What is the difference ?
What is signal receive ? Do mean receive side completion ? I would expect it is always supported, otherwise how do you recycle the memory

Reply:
TAG and AM are different APIs and we define different callbacks for it. These flags are used by both APIs. We have two different receive event types: UCT_EVENT_RECV and UCT_EVENT_RECV_SIG. If UCT_IFACE_FLAG_EVENT_RECV_SIG is supported, only UCT_EVENT_RECV_SIG event should be triggered on the receiver.

Reply:
Than maybe we should have 2 different flags for this ?
There is some issue with terminology here - signal vs completions.

Reply:
IMO, no need to consume extra cap bits and complicate API. We just define common capability bits for receive capable APIs (AM and TAG). 
This signal API is already in place and defines two types of events: 
- silent event, when receiver should not be woken up (if possible)
- event which should wake up the receiver

Reply:
@shamisp, are we ok to go, or you propose some change?
@bbenton, please have a look

Reply:
@shamisp, ping

Reply:
@shamisp, @bbenton, are we good to go?